server/docs/design/block-persistence.md

```diff
@@ -3,64 +3,213 @@
+   configurable.
+
+## Overview
+
+The design for `Block` persistence is fairly straightforward. `Block` server 
+objects should use the persistence abstractions to read, write and remove 
+`Block`s from storage, as well as resolve the paths to `Block`s. `BlockItem`s
+streamed from a producer  are read off the wire one by one and passed to an
+implementation of `BlockWriter`. The `BlockWriter` is responsible for collecting
+related `BlockItem`s into a `Block` and persisting the `Block` to storage in a
+way that  is efficient for both long-term storage and rapid retrieval at a later
+time. The `BlockWriter` is also  responsible for removing a partially written
+`Block` if an exception occurs  while writing it. For example, if half the
+`BlockItem`s of a `Block` are written  when an `IOException` occurs, the
+`BlockWriter`should remove all the `BlockItem`s of the partially written `Block`
+and pass the exception up to the  caller. Services requiring one or more `Block`
+s should leverage a `BlockReader` implementation. The `BlockReader` should be
+able to efficiently retrieve a `Block` by block number.  The `BlockReader`
+should pass unrecoverable exceptions  when reading a `Block` up to the caller.
+
+## Implementations
+
+### Block as Local Directory (a.k.a. `block-as-local-dir`)
+
+#### Purpose
+
+The purpose of this implementation is to provide a simple, local storage. Used
+mostly for testing and development purposes.
+
+#### Overview
+
+This type of storage implementation persists `Block`s to a local filesystem. A
+`Block` is persisted as a directory containing a file for each `BlockItem` that
+comprises the given `Block`. The storage has a root path where each directory
+under the root path is a given block. The names of the directories (`Block`s)
+are the respective `Block`'s `BlockNumber`.
+
+#### Configurable Parameters
+
+<!-- todo add basePath when defined -->
+- `persistence.storage.liveRootPath` - the root path where all `Block`s are
+  stored.
+
+### Block as Local File (a.k.a. `block-as-local-file`)
+
+#### Purpose
+
+The purpose of this implementation is to provide a simple, fast to resolve,
+local storage. Intended as a production default.
+
+#### Overview
+
+This type of storage implementation persists `Block`s to a local filesystem.
+A`Block` is persisted as a file containing all the `BlockItem`s that comprise
+the given `Block`. This storage options has a root directory. The root directory
+is the directory that contains all the subdirectories containing actual block
+data. A specific `Block` is stored as a file, resolved under the storage root by
+utilizing a `trie` structure. The name of the file is the respective `Block`'s
+`BlockNumber`. Optionally, the `Block` can be file is the respective `Block`'s
+`BlockNumber`. The `Block` can be compressed before being written to storage
+and, the `Block`s can be archived periodically in batches of configurable size.
+Both compression and archive are optional.
```

Comment:
Looks like we have a duplicated sentence:
```suggestion
utilizing a `trie` structure. The name of the file is the respective `Block`'s
`BlockNumber`. Each `Block` can be compressed before being written to storage,
and the `Block`s can be archived periodically in batches of configurable size.
Both compression and archive are optional.
```

Reply:
✅ Done.